capz/readme.md

```diff
@@ -4,44 +4,95 @@ These scripts and templates are used with [cluster-api-provider-azure](https://g
 
 ## Running the scripts
 
-To run these scripts locally, clone [CAPZ](https://github.com/kubernetes-sigs/cluster-api-provider-azure) and set the following environment variables (use Linux, macos, or WSL2):
+To run these scripts some setup is required.
+
+First clone and configure the following repositories
+
+### CAPZ repo setup
+
+Clone the [cluster-api-provider-azure](https://github.com/kubernetes-sigs/cluster-api-provider-azure) and checkout / sync out the latest release branch.
+
+### Cloud Provider Azure repo setup
+
+Clone the [cloud-provider-azure](https://github.com/kubernetes-sigs/cloud-provider-azure) and checkout / sync to the correct branch
+
+- if testing kubernetes@master then checkout / sync to the master branch
+- if testing a released version of Kubernetes then checkout / sync the corresponding release branch in the cloud-provider-azure repo
+
+> Note: Look at the `extra_refs` section of the **ci-kubernetes-e2e-capz-master-windows** in https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/sig-windows/release-master-windows.yaml to see which branches the SIG-Windows e2e test passes are using during the periodic jobs.
+
+### Set environment variables
+
+#### Required
 
 ```bash
 export AZURE_SUBSCRIPTION_ID=<sub-id>
 export AZURE_CLIENT_ID=<client-id>
 export AZURE_CLIENT_SECRET=<client-secret>
 export AZURE_TENANT_ID=<tenantid>
-export CAPZ_DIR="$HOME/<path-to-capz>/cluster-api-provider-azure
+export CAPZ_DIR="$HOME/<path-to-repo>/cluster-api-provider-azure"
+export AZURE_CLOUD_PROVIDER_ROOT="$HOME/<path-to-repo>/cloud-provider-azure"
+
+# run-capz-e2e.sh builds and publishes cloud-provider-azure container images
+# to the registry specified below.  Ensure you are logged in to the registry
+# prior to running the script!
+export REGISTRY="<registry>"
```

Comment:
is there anyway to use existing builds?

Reply:
I don't know yet, I can add a TODO here to investigate

Reply:
Shouldn't this be `IMAGE_REGISTRY`?

Reply:
I don't think so I set this when I run the script to make clusters
> export REGISTRY=marossetcranon.azurecr.io/cluster-api-azure

Which is an azure container registry I have set up that has anonymous read access

Reply:
So the issue I ran into was doing `export REGISTRY=quay.io/aravindh/`. The extra `/` at the end resulted in an invalid image format.